### ISSUE TITLE
problems with llhttp 9.x upgrades in v20.x

### ISSUE THREAD

**Issue by UlisesGascon** [nodejs/Release#973]
I created this issue for visibility, based on [team slack discussion](https://openjs-foundation.slack.com/archives/C045YHZC2SU/p1704738396230359)

When I was working on the release 20.11.0 proposal (step 10) I was checking the dependencies and I got an error with the `llhttp` version.

I got the version `8.1.3`, but the expected is `9.1.3` based on https://github.com/nodejs/node/pull/50080 as was included in the proposal.

The thing is that version `8.1.3` does not exist at all, as the current version is `8.1.0` since Node@20.0.0. So, the issue is that `LLHTTP_VERSION_MAJOR` didn't got updated when cherry picking this commit, only the `LLHTTP_VERSION_PATCH`.

I added the label `dont-land-on-v20` for now. But this error will occur again when landing new deps upgrades for `llhttp` in `Node@20.x`. So maybe we can work on a backport for the next release 

<details>

<summary>20.11.0 Proposal dependencies</summary>

![Screenshot 2024-01-08 at 19 25 56](https://github.com/nodejs/Release/assets/5110813/65592918-9e30-49a7-8f82-dcf0c5f9104d)

</details>

<details>

<summary>20.0.0 dependencies</summary>

![Screenshot 2024-01-08 at 19 26 12](https://github.com/nodejs/Release/assets/5110813/5c80b60d-73ad-4921-9257-ed622a1c3950)

</details>

**Comment by RafaelGSS** [comment]
cc: @shogunpanda

We should flag all v9.x updates as dont-land-on-v20.x and dont-land-on-v18.x. There's just one update for v8.x (v8.1.1) and we didn't receive an automatic update for that.

@UlisesGascon I think just dropping the commit from v20.x-staging should be enough.

**Comment by UlisesGascon** [comment]
> I wonder how https://github.com/nodejs/node/pull/50080 could be cherry-picked cleanly to v20.x-staging.

I was wondering the same, but now that I tried to cherry pick it again in staging.. I realised that I introduced the regression when I was solving the conflict in the `llhttp.h` (before the holidays). Seems like I forgot to manually patch `LLHTTP_VERSION_MAJOR` value to match the current release version. 

![Screenshot 2024-01-09 at 14 46 19](https://github.com/nodejs/Release/assets/5110813/29afaa01-93e4-40d1-937e-c494e63f00e7)

> @UlisesGascon I think just dropping the commit from v20.x-staging and rebasing again should be enough.

Yes, I dropped the commit and the versions in the binaries are as expected. So, as soon as the CI/CITGM is clear I will release the proposal.

**Next steps**

I will work in a backport for this, so it can be landed easier in the next release 👍

**Comment by RafaelGSS** [comment]
> I will work in a backport for this, so it can be landed easier in the next release 👍

There's no need for a backport for that. It's a change that should exist only on v21.x and main since it's semver-major

**Comment by ShogunPanda** [comment]
I agree. llhttp 9.1.2 was a breaking change which should not belong to v20.x.
Can you please clarify how `8.1.3` was created? Is any of the commit broken or what?

**Comment by UlisesGascon** [comment]
> Can you please clarify how `8.1.3` was created? Is any of the commit broken or what?

Seems like the issue was due a manual regression that I introduced ([see](https://github.com/nodejs/Release/issues/973#issuecomment-1883085088)). As the `llhttp@8.1.1` was included in `Node@20.10.0` when I was working in the release I didn't update the major version reference value to `9` when solving the commit conflict, so the `llhttp` library version was wrong reflected as `8.1.3` instate of `9.1.3` as expected.

**Comment by MIGUEL290891bobdrawver** [comment]
1. [ ] you
2. [ ] desorden 
3. [ ] Odette
4. [ ] desorden en parte
5. [ ] el repositor 
6. [ ] app Store iOS y Android y Android y Android y Android.

**Comment by aduh95** [comment]
Node.js 20.x is EOL
